scipy/optimize/meson.build

```diff
@@ -145,10 +135,14 @@ py3.extension_module('_bglu_dense',
   subdir: 'scipy/optimize'
 )
 
-py3.extension_module('_cython_nnls',
-  opt_gen.process('_cython_nnls.pyx'),
-  c_args: cython_c_args,
-  dependencies: np_dep,
+py3.extension_module('_slsqplib',
+  [
+    '__slsqp.h',
+    '__slsqp.c',
+    '__nnls.h',
+    '__nnls.c'
```

Comment:
Are the double underscores on the functions needed? Why not a single underscore?

Reply:
Not really any reason other than visually separating them from the `_XXX.py` files. It looked a bit unstructured when they were all garbled together. Maybe we should throw them all into a SRC dir later.

Reply:
`src/` sounds good to me.